designs/extensible-cli-and-scaffolding-plugins-phase-2.md

```diff
@@ -0,0 +1,203 @@
+# Extensible CLI and Scaffolding Plugins - Phase 2
+
+## Overview
+
+* Plugin Phase 1.5 was desgined and implemented to achieve chaining of plugins. The main goal of Phase 2 plugins is Kubebuilder discovering and using external plugins, also referred as out-of-tree plugins. So Phase 2 achieves both chaining of plugins and discovery of external plugins/source code which is not intented to be shipped with the binaries. For e.g., Kubebuilder being able to discover the helm plugin and use that without having to build the plugin with the binary. These external plugins can be implemented in any language.
+
+### Related issues and PRs
+
+* [Feature Request: Plugins Phase 2](https://github.com/kubernetes-sigs/kubebuilder/issues/1378)
+* [Extensible CLI and Scaffolding Plugins - Phase 1.5](https://github.com/kubernetes-sigs/kubebuilder/blob/master/designs/extensible-cli-and-scaffolding-plugins-phase-1-5.md)
+* [Phase 1.5 Implementation PR](https://github.com/kubernetes-sigs/kubebuilder/pull/2060)
+* [Plugin Resolution Enhancement Proposal](https://github.com/kubernetes-sigs/kubebuilder/pull/1942)
+
+### Use Cases
```

Comment:
```suggestion
### Goals
```

Reply:
Agreed. Changed it to user stories instead.